### PR TITLE
[#144] Fix typo: "Recurrencies" -> "Recurrences"

### LINKED ISSUE #144: Spelling Bug in Module Name Math.NumberTheory.Recurrencies
While "currency" pluralizes to "currencies," "recurrence" pluralizes to "recurrences."  See [Wiktionary](https://en.wiktionary.org/wiki/recurrence).  English is not my native language, so whoever misspelled this has not only my sympathy but shared pain at the intricacies of English spelling.

That said, a fix, if deemed worthwhile, should probably keep the old modules around, just importing and re-exporting the original content with some depreciation warning for some time.

### PR DESCRIPTION
**Pull request by rockbmb**
Closes #144. ~Still WIP.~

### CODE REVIEW

**Comment by rockbmb** [comment]
Ok, good to go.

**Comment by rockbmb** [comment]
@Bodigrim this isn't as trivial as the PR from earlier today that only fixes documentation, but please go over it.

File: Math/NumberTheory/Recurrences/Linear.hs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-- |
+-- Module:      Math.NumberTheory.Recurrences.Pentagonal
```
Comment (Bodigrim) [inline Math/NumberTheory/Recurrences/Linear.hs]: Module name is wrong.

File: Math/NumberTheory/Recurrences/Bilinear.hs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+-- |
+-- Module:      Math.NumberTheory.Recurrences.Pentagonal
```
Comment (Bodigrim) [inline Math/NumberTheory/Recurrences/Bilinear.hs]: Module name is wrong.

**Comment by rockbmb** [comment]
@Bodigrim good to go.

**Comment by Bodigrim** [comment]
Thanks!

**Pull request merged**